xbmc/interfaces/legacy/Player.cpp

```diff
@@ -42,12 +42,9 @@ namespace XBMCAddon
     {
       iPlayList = PLAYLIST_MUSIC;
 
-      if (_playerCore == EPC_VideoPlayer ||
-          _playerCore == EPC_MPLAYER ||
-          _playerCore == EPC_PAPLAYER)
-        playerCore = (EPLAYERCORES)_playerCore;
-      else
-        playerCore = EPC_NONE;
+      if (_playerCore != 0)
+        CLog::Log(LOGERROR, "xbmc.Player: Requested non-default player. This behavior is deprecated, plugins may no longer specify a player");
```

Comment:
I do not know what you are talking about because that api is dead for two years. Any existing plugin that still defines a player deserves to be broken. Kill backwards compatibility. End of story. 

As long as this method still works

http://mirrors.kodi.tv/docs/python-docs/15.x-isengard/xbmc.html#Player-play which just let's core handle the player and not the plugin